pkg/handler/data_recorder_pubsub.go

```diff
@@ -0,0 +1,69 @@
+package handler
+
+import (
+	"context"
+
+	"cloud.google.com/go/pubsub"
+	"github.com/checkr/flagr/pkg/config"
+	"github.com/checkr/flagr/swagger_gen/models"
+	"google.golang.org/api/option"
+
+	"github.com/sirupsen/logrus"
+)
+
+type pubsubRecorder struct {
+	enabled  bool
+	producer *pubsub.Client
+	topic    *pubsub.Topic
+}
+
+// NewPubsubRecorder creates a new Pubsub recorder
+var NewPubsubRecorder = func() DataRecorder {
+	client, err := pubsub.NewClient(
+		context.Background(),
+		config.Config.RecorderPubsubProjectID,
+		option.WithServiceAccountFile(config.Config.RecorderPubsubKeyFile),
+	)
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error getting pubsub client")
+	}
+
+	return &pubsubRecorder{
+		producer: client,
+		topic:    client.Topic(config.Config.RecorderPubsubTopicName),
+		enabled:  config.Config.RecorderEnabled,
+	}
+}
+
+func (p *pubsubRecorder) AsyncRecord(r *models.EvalResult) {
+	if !p.enabled {
+		return
+	}
+
+	pr := &pubsubEvalResult{
+		EvalResult: r,
+	}
+
+	payload, err := pr.Payload()
+	if err != nil {
+		logrus.WithField("pubsub_error", err).Error("error marshaling")
+	}
+
+	ctx := context.Background()
+	res := p.topic.Publish(ctx, &pubsub.Message{Data: payload})
+	if config.Config.RecorderPubsubVerbose {
+		id, err := res.Get(ctx)
+		if err != nil {
+			logrus.WithField("pubsub_error", err).Errorf("error pushing to pubsub: %v", id)
+		}
+	}
+}
+
+type pubsubEvalResult struct {
```

Comment:
related to https://github.com/checkr/flagr/issues/203, I think it would be great if pubsub also follow the same pattern. The better if we can fix #203 and have a single frame struct for data records.

Reply:
I totally agree, I looked at the code and tried to follow your style just to make sure everything will pass and be ok on your end... But sure I think there's a lot of place to improve some of the things.

I will take a look at those relevant issues and will update ASAP (this is night over here)

Reply:
thanks! I think you can use 
```
type pubsubMessageFrame struct {
	Payload   string `json:"payload"`
	Encrypted bool   `json:"encrypted"`
}
```

Consolidating 2 structs is the same amout of work for 3. We can fix #203 later.

Also, make sure the final payload is a JSON struct looks like
```
{
  "payload": "<json marshal of EvalResult>",
  "encrypted": false
}
```

Reply:
and test coverage of this file :)

Reply:
@zhouzhuojie regarding the pubsubMessageFrame, I added it on my end but it looks weird to marshal into json string because pubsub accepts []byte anyways so the conversion happens for no reason... I'm pushing the code to share this with you, LMK what you think

Reply:
I knew the json string payload at the first look is weird. The reason I want to standardize the log format is that 

- Extensibility. If people want to end-to-end encryption, compression, or other meta things related to the message frame, they can do it regardless of the choice of the recorder type. They can add more fields into the message frame struct. Payload as a string fits into this decision.
- Portability. The data analytics pipeline requires no changes if one wants to switch between providers.